src/domain/synthetics/tokens/useTokenAllowanceData.ts

```diff
@@ -43,15 +49,58 @@ export function useTokensAllowanceData(
           return contracts;
         }, {}),
 
-    parseResponse: (res) =>
-      Object.keys(res.data).reduce((tokenAllowance: TokensAllowanceData, address) => {
+    parseResponse: (res) => {
+      const now = Date.now();
+
+      const tokenAllowance = Object.keys(res.data).reduce((tokenAllowance: TokensAllowanceData, address) => {
         tokenAllowance[address] = res.data[address].allowance.returnValues[0];
 
         return tokenAllowance;
-      }, {} as TokensAllowanceData),
+      }, {} as TokensAllowanceData);
+
+      return {
+        tokenAllowance,
+        createdAt: now,
+      };
+    },
   });
 
+  const mergedData: TokensAllowanceData | undefined = useMemo(() => {
+    if (!spenderAddress || tokenAddresses.length === 0) {
+      return data?.tokenAllowance;
+    }
+
+    const newData: TokensAllowanceData = {};
+
+    for (const tokenAddress of tokenAddresses) {
+      const event = approvalStatuses[tokenAddress]?.[spenderAddress];
+      if (!event) {
+        if (data && tokenAddress in data.tokenAllowance) {
+          newData[tokenAddress] = data.tokenAllowance[tokenAddress];
+        }
+        continue;
+      }
+
+      const eventValue = event.value;
+      const eventCreatedAt = event.createdAt;
+
+      if (!data || !(tokenAddress in data.tokenAllowance)) {
+        newData[tokenAddress] = eventValue;
+        continue;
+      }
+
+      const allowance = data.tokenAllowance[tokenAddress];
+      if (eventCreatedAt > data.createdAt) {
+        newData[tokenAddress] = eventValue;
+      } else {
+        newData[tokenAddress] = allowance;
+      }
```

Comment:
why not simplify it down to something

const event = ...
const multicallData = ...

const dataToApply = [event, multicallData].filter(Boolean).sort(...)[0]

Reply:
simplified to 

```ts
      const event = approvalStatuses[tokenAddress]?.[spenderAddress];
      const eventValue: bigint | undefined = event?.value;
      const eventCreatedAt: number = event?.createdAt ?? 0;

      const multicallData: bigint | undefined = data?.tokenAllowance[tokenAddress];
      const multicallCreatedAt: number = data?.createdAt ?? 0;

      if (eventCreatedAt > multicallCreatedAt && eventValue !== undefined) {
        newData[tokenAddress] = eventValue;
      } else if (multicallData !== undefined) {
        newData[tokenAddress] = multicallData;
      }
```